internal/controllers/sqlitecontroller/sqlitecontroller.go

```diff
@@ -0,0 +1,63 @@
+package sqlitecontroller
+
+import (
+	"database/sql"
+	"log"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// current implementation is for sqlite
+
+func SelectStudentByID(id string) (string, error) {
+	// try opening database; if failed, throw errow
+	db, err := sql.Open("sqlite3", "./build/studtor.db")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	stmt, err := db.Prepare("SELECT student_name FROM students where student_id = ?")
+	if err != nil {
+		log.Fatal(err)
```

Comment:
I think this shouldn't be a ```Fatal``` - if there really was a fatal error, we wouldn't hit this line (line 20 would already crash). Hitting line 21 implies that the error is a value and not indicative of a program state. Hence, we should return ```nil, error``` so that the client still has something to work with

Reply:
Resolved in PR #19 . Errors for initializing database causes program exit; errors for other database methods returned to interface and handled by backend.